Frameworks/CoreGraphics/D2DWrapper.mm

```diff
@@ -19,16 +19,18 @@
 
 using namespace Microsoft::WRL;
 
-// Private helper for creating a D2DFactory
-static ComPtr<ID2D1Factory> __createD2DFactory() {
-    ComPtr<ID2D1Factory> d2dFactory;
-    FAIL_FAST_IF_FAILED(D2D1CreateFactory(D2D1_FACTORY_TYPE_SINGLE_THREADED, __uuidof(ID2D1Factory), &d2dFactory));
-    return d2dFactory;
-}
+HRESULT _CGGetD2DFactory(ID2D1Factory** factory) {
+    static ComPtr<ID2D1Factory> sFactory;
+    static HRESULT sHr;
+    static dispatch_once_t dispatchToken;
+
+    dispatch_once(&dispatchToken, ^{
```

Comment:
why did we go back to dispatch once? as mentioned before, dispatch_once doesn't buy you much except make code more dependent on  a third party library and harder to read. I preferred the second helper function / you could do something with a c++ lambda if you are concerned about the "multi-return" / "return transforming" you need to do. That is a language feature that we can be more confident in than our pseudo hacked up libdispatch just to get a static initialized.